### ISSUE TITLE
Can we collapse ContextualRegion/RegionAnnotation/RegionContainsAllele?

### ISSUE THREAD

**Issue by larrybabb** [clingen-data-model/interpretation#25]
In the spirit of MVP. Can we consider flattening ContextualRegion into RegionContainsAllele and RegionAnnotation?  See other issue which raises the idea of collapsing RegionContainsAllele into RegionAnnotation as well.

If we collapse ContextualRegion, RegionContainsAllele and RegionAnnotation we might get something like

a. region  (eg NM_000123.10:[5002..5824]) - unstructured, just formatted in a somewhat standard way.
b. type (eg hotspot)
c. containsAlleles (eg Allele1, Allele2, etc...)
d. value (? true/false - not sure what this is see RegionAnnotation.value)
e. targetAllele (eg. from RegionContainsAllele, can this be combined with c. above?)
f. value (? whether or not the targetAllele lies within the region - again can this work with d. above?)

**Comment by cbizon** [comment]
RegionAnnotation.containsAlleles, I think, was a left-over from when we split these concepts apart.

I think that I'm ok with combining them again, but I'm not sure about ContextualRegion.  We certainly can do it, but I don't really like defining an entity and saying something about then entity all in one un-breakable unit.    If everybody liked the simplicity, I could live with it.

The other thing to consider here is the provenance of these things.   If we combine the statement about what the region is (it is functional) and what alleles are in the region, then we lose independent provenance for those two things.   I suspect that they come from different places.  The allele in the region is fairly simple though, and maybe doesn't require a full prov?

**Comment by larrybabb** [comment]
Chris proposal - merge regionAnnotation & containsAlleles into regionAnnotation. Leave contextualRegion.
Bradford - ? will respond.

**Comment by larrybabb** [comment]
@cbizon @bpow 
After re-thinking this and bringing it up on the call today, I think I would like to see ContextualRegion get collapsed into RegionAnnotation overall.  Since Chris is fine with merging RegionAnnotation with RegionContainsAlleles, we should end up with one (large, but simpler - in a way) structure.

@cbizon makes the point at the end of his last post that 
The allele in the region is fairly simple though, and maybe doesn't require a full prov?
I would agree with that. At this point it is hard to imagine many systems for the foreseeable future will capture this level of detail let alone the provenance. And, when they do, is it really that significant to the goals of the data?

So, I would vote for collapsing this entire structure.

@bpow please weigh in.  

I will do the data work to transform the examples and worksheet.

**Comment by bpow** [comment]
Yep, this seems pragmatic especially for MVP. We can defer the additional complexity until someone provides a good use case that we really need it.

**Comment by cbizon** [comment]
OK, I agree. Let's combine all.

**Comment by rrfreimuth** [comment]
I wanted to review the model prior to commenting on this thread, but my VP license just expired.  I put in a request to get it extended but am high and dry in the meantime.

**Comment by larrybabb** [comment]
If you only need review capabilities then I can re-setup the view only access which is free. Let me know if that is the direction you want to go.

**Comment by bpow** [comment]
@larrybabb : it looks like you are in the process of working on this... Dangling references to no-longer existent entities (E42) and attributes (A177-179, and others) are breaking the build process.

I cleared the rows in `Data` referencing E42, but then saw the same problem with attributes, and didn't know if you are leaving these cells partially filled to keep track of what you are doing.

For now, I will try to put some error handling code in `reformat_examples.rb`.

**Comment by bpow** [comment]
As I start to see what you are doing here (and the name of the new Type: `RegionTypesAndAlleles`), I'm beginning to have second thoughts about what exactly we are modeling here.

It seems to me that there is a concept of a Region, and a separate concept of whether that Region contains an allele...

If we define a `RegionTypesAndAlleles` object as something like 'zinc finger domain from position 28-54 of protein Z', and know of a few alleles that are in that region... when we find another allele, wouldn't we have to make a whole new `RegionTypesAndAlleles` object? Pinging @cbizon for thoughts.

So, I'm still OK with merging `ContextualRegion` and `RegionAnnotation`, but think that `RegionContainsAllele` is a statement (`EvidenceLine`?) with attributes region, allele (and possibly a boolean...)

At any rate, references to `RegAnno`s and `CtxReg`s in _DataAttribute need to be cleaned up, for now they are breaking referential integrity (and by extension diagramming code...)

**Comment by larrybabb** [comment]
Completed. RegionAnnotation and RegionContainsAllele has been combined into "Region". In addition, the ContextualRegion has been merged into Region and BenignMissenseVariationRegion.

**Comment by bpow** [comment]
For future reference (not re-opening this or opening a new issue yet...), I still think that the `includedAlleles`doesn't belong to the merged Region type as a property...

We have no way of tracking versions of entities. I had presumed (though I don't know if we have discussed) that entities would be immutable (a modified entity gets a new id..., perhaps derived from an old one). There could potentially be a lot of churn of `RegionTypesAndAlleles` as we add discovered alleles.

If we don't keep track of prior versions or assign new ids for immutable instances, then we lose trackablility/provenance of data available at the time an assertion was made.
